File: backend/app/constants/object_type.py
"""Canonical object_type strings for polymorphic links (API + DB)."""

from enum import Enum


class ObjectType(str, Enum):
    company = "company"
    contact = "contact"
    lead = "lead"
    manufacturing_partner = "manufacturing_partner"
    product = "product"
    rfq = "rfq"
    quotation = "quotation"
    sample = "sample"
    order = "order"
    field_visit = "field_visit"
    field_visit_plan = "field_visit_plan"
    task = "task"
    interaction = "interaction"
    user = "user"
    file = "file"
    market_intelligence = "market_intelligence"


ALLOWED_OBJECT_TYPES = frozenset(m.value for m in ObjectType)


def normalize_object_type(value: str) -> str:
    v = value.strip().lower()
    if v not in ALLOWED_OBJECT_TYPES:
        raise ValueError(f"Invalid object_type: {value}")
    return v

File: frontend/src/constants/objectType.ts
/** 与后端 `app.constants.object_type.ObjectType` 对齐 */
export const OBJECT_TYPES = [
  'company',
  'contact',
  'lead',
  'manufacturing_partner',
  'product',
  'rfq',
  'quotation',
  'sample',
  'order',
  'field_visit',
  'field_visit_plan',
  'task',
  'interaction',
  'user',
  'file',
  'market_intelligence',
] as const

export type ObjectType = (typeof OBJECT_TYPES)[number]

const SET = new Set<string>(OBJECT_TYPES)

export function normalizeObjectType(raw: string): ObjectType {
  const v = raw.trim().toLowerCase()
  if (!SET.has(v)) {
    throw new Error(`Invalid object_type: ${raw}`)
  }
  return v as ObjectType
}
